tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
++ **girder_job_title** This changes the job title, by default (e.g if girder\_job\_title is omitted) it is "\<unnamed job\>".
++ **girder_job_type** This changes the job type, by default it is "celery"
++ **girder_job_public** This changes the public status of the job, by default it is False
++ **girder_job_handler** This changes the job handler,  by default it is "celery_handler" (Note: do not change this unless you are sure you know what you are doing).
++ **girder_job_other_fields** This field allows you to set additional handler specific fields on the Job.  Currently the "celery_handler" has no additional standard fields. This could however be used to extend the job model data to suit application specific needs.
++ **girder_user** This field sets the girder user which belongs to the job.  It should be a full user model. By default it is the return value of ```girder.api.rest.getCurrentUser()```
+
+Here is an example of using these fields:
+
+```python
+# Import the Celery Task object
+from my_custom_package.tasks import fibonacci
+
+...
+
+# Generate a message and place it on the queue
+async_result = fibonacci.delay(20, 
+    girder_job_title='Fibonacci Job',
+	girder_job_type='my_custom_type',
+	girder_job_public=True,
+	girder_job_other_fields= {
+	    'some_custom_field': 'some_custom_value'
+	})
+
+# Return the Girder Job
+return async_result.job
+```
+
+These keyword arguments may also be passed as keyword arguments (or 'options') to the ```.apply_async(...)```, ```.s(...)``` and ```.signature(...)``` methods:
+
+```python
+async_result = fibonacci.apply_async(
+	args=(20,), 
+	kwargs={}, 
+    girder_job_title='Fibonacci Job',
+	girder_job_type='my_custom_type',
+	girder_job_public=True,
+	girder_job_other_fields= {
+	    'some_custom_field': 'some_custom_value'
+	})
+
+```
+
+Or equally:
+
+```python
+sig = fibonacci.s(20, 
+    girder_job_title='Fibonacci Job',
+	girder_job_type='my_custom_type',
+	girder_job_public=True,
+	girder_job_other_fields= {
+	    'some_custom_field': 'some_custom_value'
+	})
+	
+async_result = sig.delay()
+
+return async_result.job
+```
+
+
+#### Job Defaults
+
+The above keyword arguments can be used to set per-invocation values for the generated job model. While this is useful for specific invocations, generally Girder Job metadata will be similar across task invocations.  Default values for this metadata may be set by decorating the task definition with the ```girder_job``` decorator:
+
+```python
+from girder_worker.utils import girder_job
+from girder_worker.app import app
+
+
+@girder_job(title='Fibonacci Job',
```

Comment:
No unfortunately order is important here. I have added a line to the docs to make this clear.